### PR TITLE
Added Save Image option 

### PR DESCRIPTION
**Pull request by Flxp49**
Hi there👋

#10

- Save Image option:  Save images imported by recipes. If no location is specified, will follow obsidian setting: Files and links > new attachment location. Note: if settings is toggled off or image save fails, will revert to direct URL.
- Changed the default template from `{{{name}}}` to `![[{{image}}]]`

Save image is by default enabled and set to follow obsidian's new attachment location.

### CODE REVIEW

**Comment by seethroughdev** [comment]
Thanks again @Flxp49 for doing this! I'll dig through the code in a bit here. Can I also make you a maintainer on this project?

**Comment by seethroughdev** [comment]
Overall, looks great. Added a couple of small notes.

**Review by seethroughdev — APPROVED** [review]
All really small stuff. Overall, looks great, and I know people will use the feature. Thank you again!

File: src/constants.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ url: {{url}}
 
 {{{description}}}
 
-![{{{name}}}]({{image}})
+![[{{image}}]]
```
[the remark this answers is not in the document]
Comment (Flxp49) [reply src/constants.ts:19]: Download is enabled by default hence the change. I will add a section ( regarding the alt tag for direct links ) in README for user convenience.

File: src/main.ts [hunk 2]
```diff
@@ -13,10 +13,12 @@ import {
 	requestUrl,
 	normalizePath,
 	TFolder,
+	TFile,
 } from "obsidian";
 import * as handlebars from "handlebars";
 import type { Recipe } from "schema-dts";
 import * as cheerio from "cheerio";
+import { fileTypeFromBuffer } from 'file-type';
```
[the remark this answers is not in the document]
Comment (Flxp49) [reply src/main.ts:21]: We do need it to detect the type of image from the arraybuffer. We could implement our own function but would require a lot work from scratch.

File: src/settings.ts [hunk 3]
```diff
@@ -40,7 +44,7 @@ export class SettingsTab extends PluginSettingTab {
 				text.setPlaceholder("eg: Recipes")
 					.setValue(this.plugin.settings.folder)
 					.onChange(async (value) => {
-						this.plugin.settings.folder = value;
+						this.plugin.settings.folder = value.trim();
```
[the remark this answers is not in the document]
Comment (Flxp49) [reply src/settings.ts:47]: Trim() is for when the user leaves whitespaces by mistake. The value when blank is `""`.

**Review by seethroughdev — APPROVED** [review]
These changes look great. Sorry for the delay!

**Comment by seethroughdev** [comment]
You okay for me to merge @Flxp49 ?

**Comment by Flxp49** [comment]
> You okay for me to merge @Flxp49 ?

Yep you can go ahead.

**Pull request merged**